python/tvm/relax/op/base.py

```diff
@@ -253,6 +254,19 @@ def render_object(val: tvm.Object) -> str:
     return str(val)
 
 
+@tvm.register_func("relax.run.shape_to_tensor")
+def relax_shape_to_tensor(shape_tuple: tvm.runtime.ShapeTuple) -> tvm.nd.NDArray:
+    """
+    Takes a ShapeTuple and convert it to NDArray.
+
+    Parameters
+    ----------
+    shape_tuple: tvm.runtime.ShapeTuple
+        Shape tuple that we want to convert to NDArray at runtime
+    """
+    return tvm.nd.array([int(v) for v in shape_tuple])
```

Comment:
do we assume it's always on cpu?

Reply:
Yes. shape tuple and shape computation happen on the CPU side.